src/Microsoft.AspNetCore.Razor.Design/build/netstandard2.0/Microsoft.AspNetCore.Razor.Design.targets

```diff
@@ -0,0 +1,238 @@
+      Condition="!Exists('%(_RazorGenerated.RelativeDir)')" />
+
+    <RazorGenerate
+      Debug="$(_RazorDebugGenerateCodeTask)"
+      DebugTool="$(_RazorDebugGenerateCodeTool)"
+      ToolAssembly="$(_RazorGenerateToolAssembly)"
+      Sources="@(RazorCompile)"
+      ProjectRoot="$(MSBuildProjectDirectory)"
+      TagHelperManifest="$(_RazorTagHelperOutputCache)"
+      OutputPath="$(RazorGenerateOutputPath)" />
+
+    <ItemGroup>
+      <FileWrites Include="@(_RazorGenerated)" />
+    </ItemGroup>
+  </Target>
+
+  <!--
+    What follows here was copied and modified from the XamlPreCompile target in 
+    Microsoft.CSharp.CurrentVersion.targets
+
+    The XamlPreCompile target must remain identical to
+    the CoreCompile target in Microsoft.CSharp.Core.targets.
+    Any updates to one must be made to the other.
+  -->
+  <Target
+    Name="RazorCoreCompile"
+    Inputs="$(MSBuildAllProjects);
+            @(_RazorGenerated);
+            @(_CoreCompileResourceInputs);
+            $(ApplicationIcon);
+            $(AssemblyOriginatorKeyFile);
+            @(RazorReferencePath);"
+              
+    Outputs="$(_RazorIntermediateAssembly)"
+    Returns=""
+    DependsOnTargets="_RazorResolveTagHelpers;RazorCoreGenerate">
+
+    <!-- These two compiler warnings are raised when a reference is bound to a different version
+          than specified in the assembly reference version number.  MSBuild raises the same warning in this case,
+          so the compiler warning would be redundant. -->
+    <PropertyGroup Condition="('$(TargetFrameworkVersion)' != 'v1.0') and ('$(TargetFrameworkVersion)' != 'v1.1')">
+      <NoWarn>$(NoWarn);1701;1702</NoWarn>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <!-- To match historical behavior, when inside VS11+ disable the warning from csc.exe indicating that no sources were passed in-->
+      <NoWarn Condition=" '$(BuildingInsideVisualStudio)' == 'true' and '$(VisualStudioVersion)' != '' and '$(VisualStudioVersion)' > '10.0' ">$(NoWarn);2008</NoWarn>
+    </PropertyGroup>
+
+    <ItemGroup Condition="'$(TargetingClr2Framework)'=='true'">
+      <ReferencePath>
+        <EmbedInteropTypes/>
+      </ReferencePath>
+    </ItemGroup>
+
+    <PropertyGroup>
+      <!-- If the user has specified AppConfigForCompiler, we'll use it. If they have not, but they set UseAppConfigForCompiler,
+              then we'll use AppConfig -->
+      <AppConfigForCompiler Condition="'$(AppConfigForCompiler)' == '' and '$(UseAppConfigForCompiler)' == 'true'">$(AppConfig)</AppConfigForCompiler>
+    </PropertyGroup>
+
+    <!-- Prefer32Bit was introduced in .NET 4.5. Set it to false if we are targeting 4.0 -->
+    <PropertyGroup Condition="('$(TargetFrameworkVersion)' == 'v4.0')">
+      <Prefer32Bit>false</Prefer32Bit>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'$(UseSharedCompilation)' == ''">
+      <UseSharedCompilation>true</UseSharedCompilation>
+    </PropertyGroup>
+
+    <Csc
```

Comment:
We should probably add `Deterministic` to this. Are there any other parameters that were left off intentionally, or was this just a copy-paste? I'm mildly concerned about the future-proofness of using this task directly from an NuGet package and keepign it aligned with parameters available from https://github.com/dotnet/roslyn/blob/version-2.3.2/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets.

Also, we should probably also consider how to enable developers who really care about determinism, even in symbol generation. To enable that, we'd need to provide a value for the `SourceLink` and `PathMap`  parameters that makes symbol contents insensitive to the folder path containing the source code. Example: https://github.com/dotnet/roslyn/blob/5c3dff5ba9c7905b8b194378acca7f0b288e366e/build/Targets/Settings.props#L124-L136

Reply:
This is a copy paste. We need to talk to the relevant Roslyn folks about all of this.

Reply:
https://github.com/aspnet/Razor/issues/1845